Patch makefile with full-clean feature

reactive-firewall opened the pull request:
* minor change to makefile for development cycle automation.

dependabot[bot] committed:
Bump certifi from 2025.6.15 to 2025.7.9 (#50)

Bumps [certifi](https://github.com/certifi/python-certifi) from 2025.6.15 to 2025.7.9.
- [Commits](https://github.com/certifi/python-certifi/compare/2025.06.15...2025.07.09)

---
updated-dependencies:
- dependency-name: certifi
  dependency-version: 2025.7.9
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump types-docutils from 0.21.0.20250708 to 0.21.0.20250710 (#51)

Bumps [types-docutils](https://github.com/typeshed-internal/stub_uploader) from 0.21.0.20250708 to 0.21.0.20250710.
- [Commits](https://github.com/typeshed-internal/stub_uploader/commits)

---
updated-dependencies:
- dependency-name: types-docutils
  dependency-version: 0.21.0.20250710
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

reactive-firewall committed:
Small improvements to Makefile for more agnostic development (- WIP #49 -)

dependabot[bot] committed:
Bump certifi from 2025.7.9 to 2025.7.14 (#55)

Bumps [certifi](https://github.com/certifi/python-certifi) from 2025.7.9 to 2025.7.14.
- [Commits](https://github.com/certifi/python-certifi/compare/2025.07.09...2025.07.14)

---
updated-dependencies:
- dependency-name: certifi
  dependency-version: 2025.7.14
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

reactive-firewall committed:
Update Makefile

Co-authored-by: Stefan <96178532+stefan6419846@users.noreply.github.com>

dependabot[bot] committed:
Bump types-docutils from 0.21.0.20250710 to 0.21.0.20250715 (#58)

Bumps [types-docutils](https://github.com/typeshed-internal/stub_uploader) from 0.21.0.20250710 to 0.21.0.20250715.
- [Commits](https://github.com/typeshed-internal/stub_uploader/commits)

---
updated-dependencies:
- dependency-name: types-docutils
  dependency-version: 0.21.0.20250715
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump mypy from 1.16.1 to 1.17.0 (#57)

Bumps [mypy](https://github.com/python/mypy) from 1.16.1 to 1.17.0.
- [Changelog](https://github.com/python/mypy/blob/master/CHANGELOG.md)
- [Commits](https://github.com/python/mypy/compare/v1.16.1...v1.17.0)

---
updated-dependencies:
- dependency-name: mypy
  dependency-version: 1.17.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump types-pygments from 2.19.0.20250516 to 2.19.0.20250715 (#56)

Bumps [types-pygments](https://github.com/typeshed-internal/stub_uploader) from 2.19.0.20250516 to 2.19.0.20250715.
- [Commits](https://github.com/typeshed-internal/stub_uploader/commits)

---
updated-dependencies:
- dependency-name: types-pygments
  dependency-version: 2.19.0.20250715
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

reactive-firewall committed:
Split full-clean and un-setup into two

dependabot[bot] committed:
Bump nh3 from 0.2.21 to 0.2.22 (#60)

Bumps [nh3](https://github.com/messense/nh3) from 0.2.21 to 0.2.22.
- [Release notes](https://github.com/messense/nh3/releases)
- [Commits](https://github.com/messense/nh3/compare/v0.2.21...v0.2.22)

---
updated-dependencies:
- dependency-name: nh3
  dependency-version: 0.2.22
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

pkwasniok committed:
Add --collect-all-failures docs to USAGE.md (#61)

Closes #59.

reactive-firewall commented:
@codecov-ai-reviewer review

ghost commented:
On it! We are reviewing the PR and will provide feedback shortly.

ghost commented:
## PR Description

This pull request improves the project's development environment and adds a new feature to control license failure reporting. It enhances the robustness of the Makefile, updates development dependencies, and introduces the `collect-all-failures` option for the `pip-licenses` tool.

<details>
<summary><b>Click to see more</b></summary>

### Key Technical Changes
The key technical changes include: 1) Modifying the Makefile to determine the repository name from the local directory instead of relying on the remote URL, improving reliability. 2) Adding a `venv` target to the Makefile for virtual environment creation. 3) Introducing a `full-clean` target to remove various cached files and directories, and an `un-setup` target to remove the virtual environment. 4) Updating development dependencies in `requirements/dev.txt` to newer versions. 5) Adding documentation for the `collect-all-failures` option in `USAGE.md`, which allows the tool to collect all license failures instead of stopping at the first one.

### Architecture Decisions
The architectural decisions primarily involve improvements to the build and development process. The change in how the repository name is determined aims for greater reliability in different development environments. The addition of cleaning targets provides better control over the project's state during development. The `collect-all-failures` option introduces a new mode of operation for the `pip-licenses` tool, allowing for more comprehensive license compliance checks.

### Dependencies and Interactions
The changes primarily affect the development environment and the `pip-licenses` tool's behavior. The updated dependencies in `requirements/dev.txt` may have implications for the tool's functionality and compatibility. The Makefile changes interact with the virtual environment and project cleaning processes. The `collect-all-failures` option directly impacts how license failures are reported by the tool.

### Risk Considerations
Potential risks include: 1) The updated dependencies in `requirements/dev.txt` may introduce compatibility issues or unexpected behavior. Thorough testing is required to ensure the tool functions correctly with the new dependency versions. 2) The `un-setup` target in the Makefile removes the virtual environment, which could lead to accidental data loss if not used carefully. A warning message has been suggested to mitigate this risk. 3) The `collect-all-failures` option may reveal a large number of license failures that were previously hidden, requiring further investigation and resolution.

### Notable Implementation Details
Notable implementation details include: 1) The use of `git rev-parse --show-toplevel` in the Makefile to determine the repository name. 2) The addition of error suppression (`2>/dev/null || :`) in the Makefile cleaning targets. 3) The introduction of the `collect-all-failures` option and its corresponding documentation in `USAGE.md`.
</details>

reactive-firewall commented on Makefile:
```diff
@@ -62,3 +71,18 @@ build: clean
+	rm -vfRd ./piplicenses/__pycache__ 2>/dev/null || : ;
+	rm -vfRd ./.coverage 2>/dev/null || : ;
+	rm -vfRd ./.mypy_cache 2>/dev/null || : ;
+	rm -vfRd ./.pytest_cache 2>/dev/null || : ;
```
```suggestion
	rm -vrf *.egg-info || : ;
	rm -vfrd ./piplicenses/__pycache__ 2>/dev/null || : ;
	rm -vfrd ./.coverage 2>/dev/null || : ;
	rm -vfrd ./.mypy_cache 2>/dev/null || : ;
	rm -vfrd ./.pytest_cache 2>/dev/null || : ;
```

reactive-firewall commented on Makefile:
```diff
@@ -62,3 +71,18 @@ build: clean
+	rm -vfRd ./.mypy_cache 2>/dev/null || : ;
+	rm -vfRd ./.pytest_cache 2>/dev/null || : ;
+
+# hard reset
+.PHONY: un-setup
+un-setup:: full-clean
+	rm -vfRd ./venv 2>/dev/null || : ;
```
```suggestion
# hard reset
.PHONY: un-setup
un-setup:: full-clean
	rm -vfrd ./venv 2>/dev/null || : ;
```

reactive-firewall committed:
Apply suggestions from AI code review

Namely change -R to -r for increased portability across different versions of `rm`

Signed-off-by: Mr. Walls <reactive-firewall@users.noreply.github.com>

dependabot[bot] committed:
Bump nh3 from 0.2.22 to 0.3.0 (#63)

Bumps [nh3](https://github.com/messense/nh3) from 0.2.22 to 0.3.0.
- [Release notes](https://github.com/messense/nh3/releases)
- [Commits](https://github.com/messense/nh3/compare/v0.2.22...v0.3.0)

---
updated-dependencies:
- dependency-name: nh3
  dependency-version: 0.3.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump license-expression from 30.4.3 to 30.4.4 (#64)

Bumps [license-expression](https://github.com/aboutcode-org/license-expression) from 30.4.3 to 30.4.4.
- [Release notes](https://github.com/aboutcode-org/license-expression/releases)
- [Changelog](https://github.com/aboutcode-org/license-expression/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/aboutcode-org/license-expression/compare/v30.4.3...v30.4.4)

---
updated-dependencies:
- dependency-name: license-expression
  dependency-version: 30.4.4
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump types-docutils from 0.21.0.20250715 to 0.21.0.20250722 (#65)

Bumps [types-docutils](https://github.com/typeshed-internal/stub_uploader) from 0.21.0.20250715 to 0.21.0.20250722.
- [Commits](https://github.com/typeshed-internal/stub_uploader/commits)

---
updated-dependencies:
- dependency-name: types-docutils
  dependency-version: 0.21.0.20250722
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump rich from 14.0.0 to 14.1.0 (#66)

Bumps [rich](https://github.com/Textualize/rich) from 14.0.0 to 14.1.0.
- [Release notes](https://github.com/Textualize/rich/releases)
- [Changelog](https://github.com/Textualize/rich/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Textualize/rich/compare/v14.0.0...v14.1.0)

---
updated-dependencies:
- dependency-name: rich
  dependency-version: 14.1.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump coverage[toml] from 7.9.2 to 7.10.0 (#67)

Bumps [coverage[toml]](https://github.com/nedbat/coveragepy) from 7.9.2 to 7.10.0.
- [Release notes](https://github.com/nedbat/coveragepy/releases)
- [Changelog](https://github.com/nedbat/coveragepy/blob/master/CHANGES.rst)
- [Commits](https://github.com/nedbat/coveragepy/compare/7.9.2...7.10.0)

---
updated-dependencies:
- dependency-name: coverage[toml]
  dependency-version: 7.10.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

stefan6419846 committed:
Improve modularization (#69)

Closes #2.

reactive-firewall committed:
Apply changes from review PR #54

Changes:

* expand cache cleanup to tests in full-clean

* STYLE: consistant suppression of stderr in full-clean
* STYLE: remove some comments and use consistant spacing in Makefile

dependabot[bot] committed:
Bump types-docutils from 0.21.0.20250722 to 0.21.0.20250728 (#70)

Bumps [types-docutils](https://github.com/typeshed-internal/stub_uploader) from 0.21.0.20250722 to 0.21.0.20250728.
- [Commits](https://github.com/typeshed-internal/stub_uploader/commits)

---
updated-dependencies:
- dependency-name: types-docutils
  dependency-version: 0.21.0.20250728
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump coverage[toml] from 7.10.0 to 7.10.1 (#71)

Bumps [coverage[toml]](https://github.com/nedbat/coveragepy) from 7.10.0 to 7.10.1.
- [Release notes](https://github.com/nedbat/coveragepy/releases)
- [Changelog](https://github.com/nedbat/coveragepy/blob/master/CHANGES.rst)
- [Commits](https://github.com/nedbat/coveragepy/compare/7.10.0...7.10.1)

---
updated-dependencies:
- dependency-name: coverage[toml]
  dependency-version: 7.10.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

reactive-firewall commented:
@codecov-ai-reviewer review

ghost commented:
On it! We are reviewing the PR and will provide feedback shortly.

pkwasniok committed:
Licenses filtering doesn't work properly when list ends with semicolon and partial matching is enabled (#68)

Closes #62.

---------

Co-authored-by: Patryk Kwasniok <pkwasniok@kplabs.pl>

stefan6419846 committed:
version 2.0.0

dependabot[bot] committed:
Bump docutils from 0.21.2 to 0.22 (#72)

Bumps [docutils](https://github.com/rtfd/recommonmark) from 0.21.2 to 0.22.
- [Changelog](https://github.com/readthedocs/recommonmark/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rtfd/recommonmark/commits)

---
updated-dependencies:
- dependency-name: docutils
  dependency-version: '0.22'
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump mypy from 1.17.0 to 1.17.1 (#73)

Bumps [mypy](https://github.com/python/mypy) from 1.17.0 to 1.17.1.
- [Changelog](https://github.com/python/mypy/blob/master/CHANGELOG.md)
- [Commits](https://github.com/python/mypy/compare/v1.17.0...v1.17.1)

---
updated-dependencies:
- dependency-name: mypy
  dependency-version: 1.17.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump pip-tools from 7.4.1 to 7.5.0 (#74)

Bumps [pip-tools](https://github.com/jazzband/pip-tools) from 7.4.1 to 7.5.0.
- [Release notes](https://github.com/jazzband/pip-tools/releases)
- [Changelog](https://github.com/jazzband/pip-tools/blob/main/CHANGELOG.md)
- [Commits](https://github.com/jazzband/pip-tools/compare/7.4.1...v7.5.0)

---
updated-dependencies:
- dependency-name: pip-tools
  dependency-version: 7.5.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

reactive-firewall committed:
Removed stray comment from Makefile as suggested (See PR #54)

Co-authored-by: stefan6419846 <stefan6419846@users.noreply.github.com>
Signed-off-by: stefan6419846 <stefan6419846@users.noreply.github.com>
